src/Init/System/IO.lean

```diff
@@ -31,37 +31,61 @@ def IO.RealWorld : Type := Unit
 -/
 def EIO (ε : Type) : Type → Type := EStateM ε IO.RealWorld
 
-@[inline] def EIO.catchExceptions (x : EIO ε α) (h : ε → EIO Empty α) : EIO Empty α :=
-  fun s => match x s with
-  | EStateM.Result.ok a s     => EStateM.Result.ok a s
-  | EStateM.Result.error ex s => h ex s
-
 instance : Monad (EIO ε) := inferInstanceAs (Monad (EStateM ε IO.RealWorld))
 instance : MonadFinally (EIO ε) := inferInstanceAs (MonadFinally (EStateM ε IO.RealWorld))
 instance : MonadExceptOf ε (EIO ε) := inferInstanceAs (MonadExceptOf ε (EStateM ε IO.RealWorld))
 instance : OrElse (EIO ε α) := ⟨MonadExcept.orElse⟩
 instance [Inhabited ε] : Inhabited (EIO ε α) := inferInstanceAs (Inhabited (EStateM ε IO.RealWorld α))
 
+/-- An `EIO` monad that cannot throw exceptions. -/
+abbrev RealM := EIO Empty
+
+@[inline] def RealM.toEIO (self : RealM α) : EIO ε α :=
+  fun s => match self s with
+  | EStateM.Result.ok a s => EStateM.Result.ok a s
+
+instance : MonadLift RealM (EIO ε) := ⟨RealM.toEIO⟩
+
+@[inline] def EIO.toRealM (self : EIO ε α) : RealM (Except ε α) :=
+  fun s => match self s with
+  | EStateM.Result.ok a s     => EStateM.Result.ok (Except.ok a) s
+  | EStateM.Result.error ex s => EStateM.Result.ok (Except.error ex) s
+
+@[inline] def EIO.toRealM_ (self : EIO ε α) : RealM Unit :=
```

Comment:
I don't think we use this naming style. I'm also not sure we should have this function.

Reply:
@Kha This naming style is copied from Haskell. I currently have a minimum make do API for [`RealM`](https://github.com/leanprover/lake/blob/b2709a91741df37f2ee6d4af74eb9ceb63f88ae2/Lake/RealM.lean) in Lake and this was the last function missing from the introduction of `RealM` here that I use there. Thus, I stuck it here. I am happy to revert the change if you think it is inappropriate, but I can say, at the very less, it is useful to me (in Lake),

Reply:
It looks like a huge red flag to me. When is it ok to swallow IO exceptions?

Reply:
@Kha when performing an action where you can't reasonably handle the error and/or you don't really care if it completes.

For example, printing out log messages. If the print errors, then the error itself can't be logged (because logging is the thing that is erroring). You also generally don't want the whole program to stop because you failed to print a log line. Thus, it is fine to swallow the IO error.

Reply:
We are already more strict about discarding return values in `do` than Haskell, so I don't think it's consistent to provide functions like this one. I don't like the "cute" naming style either. We have `discard` (or `let _ <-`) for this, let's be explicit when we use it.

Reply:
@Kha I removed the function